include/dali/core/error_handling.h

```diff
@@ -221,27 +215,25 @@ inline dali::string GetStacktrace() {
 #define DALI_ENFORCE_VALID_INDEX(var, upper) \
   DALI_ENFORCE_IN_RANGE(var, 0, upper)
 
+#define DALI_STR2(x) #x
+#define DALI_STR(x) DALI_STR2(x)
+#define FILE_AND_LINE __FILE__ ":" DALI_STR(__LINE__)
+
+#define DALI_MESSAGE(str)\
+  (std::string("[" FILE_AND_LINE "] ") + str + dali::GetStacktrace())
+
+#define DALI_FAIL(str)                            \
+    throw dali::DALIException(DALI_MESSAGE(str)); \
```

Comment:
this could be a function

Reply:
Nope. __FILE__ and __LINE__ will lose their meaning.